client/my-sites/site-settings/section-import.jsx

```diff
@@ -16,31 +18,26 @@ import MediumImporter from 'my-sites/importer/importer-medium';
 import { fetchState } from 'lib/importer/actions';
 import { appStates, WORDPRESS, MEDIUM } from 'state/imports/constants';
 import EmailVerificationGate from 'components/email-verification/email-verification-gate';
+import { getSelectedSite } from 'state/ui/selectors';
 
-export default React.createClass( {
-	displayName: 'SiteSettingsImport',
+class SiteSettingsImport extends Component {
+	static propTypes = {
+		site: PropTypes.object,
+	};
 
-	propTypes: {
-		site: React.PropTypes.oneOfType( [
-			React.PropTypes.bool,
-			PropTypes.shape( {
-				slug: PropTypes.string.isRequired,
-				title: PropTypes.string.isRequired
-			} )
-		] )
-	},
+	constructor( props ) {
+		super( props );
 
-	componentDidMount: function() {
+		this.state = getImporterState();
```

Comment:
What you describe @sirreal comes probably from [here](https://babeljs.io/blog/2015/06/07/react-on-es6-plus), which has seen increased adoption as a Calypso style guide of sorts. (FWIW, I prefer `state = ` over the constructor approach, too.)